src/Components/CreateImageWizard/steps/Locale/components/LanguagesDropDown.tsx

```diff
@@ -1,9 +1,175 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { FormGroup } from '@patternfly/react-core';
+import {
+  Button,
+  Chip,
+  ChipGroup,
+  FormGroup,
+  MenuToggle,
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+
+import { useAppDispatch, useAppSelector } from '../../../../../store/hooks';
+import {
+  addLanguage,
+  removeLanguage,
+  selectLanguages,
+} from '../../../../../store/wizardSlice';
+import { languagesList } from '../languagesList';
 
 const LanguagesDropDown = () => {
-  return <FormGroup isRequired={false} label="Languages"></FormGroup>;
+  const languages = useAppSelector(selectLanguages);
+  const dispatch = useAppDispatch();
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [filterValue, setFilterValue] = useState<string>('');
+  const [selectOptions, setSelectOptions] = useState<string[]>(languagesList);
+
+  useEffect(() => {
+    let filteredLanguages = languagesList;
+
+    if (filterValue) {
+      filteredLanguages = languagesList.filter((language: string) =>
+        String(language).toLowerCase().includes(filterValue.toLowerCase())
+      );
+      if (!filteredLanguages.length) {
+        filteredLanguages = [`No results found for "${filterValue}"`];
+      }
+      if (!isOpen) {
+        setIsOpen(true);
+      }
+    }
+    setSelectOptions(filteredLanguages.sort((a, b) => sortfn(a, b)));
+
+    // This useEffect hook should run *only* on when the filter value changes.
+    // eslint's exhaustive-deps rule does not support this use.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [filterValue]);
+
+  const sortfn = (a: string, b: string) => {
```

Comment:
Need to update this so it uses the reusable utility function. Will take care of it tomorrow.

Reply:
Added a commit for the reusable function, thank you!